lib/src/js/value/color.dart

```diff
@@ -72,7 +372,68 @@ final JSClass colorClass = () {
 ///
 /// This ensures that an explicitly null alpha will be treated as a missing
 /// component.
-num? _handleUndefinedAlpha(num? alpha) => isUndefined(alpha) ? 1 : alpha;
+double? _handleUndefinedAlpha(double? alpha) => isUndefined(alpha) ? 1 : alpha;
+
+/// This procedure takes a `channel` name, an object `changes` and a SassColor
+/// `initial` and returns the result of applying the change for `channel` to
+/// `initial`.
+double? _changeComponentValue(
+    SassColor initial, String channel, Map<String, double?> changes) {
+  var initialValue = initial.channel(channel);
+  if (!changes.containsKey(channel)) {
+    return initialValue;
+  }
+  return changes[channel];
+}
```

Comment:
I think you can more cleanly write this as

```suggestion
        SassColor initial, String channel, Map<String, double?> changes) =>
  changes.containsKey(channel) ? changes[channel] : initial.channel(channel);
```

this also avoids assigning `initialValue` in the case where it's not used.